src/server/v2.0/handler/artifact.go

```diff
@@ -36,15 +38,61 @@ func newArtifactAPI() *artifactAPI {
 	return &artifactAPI{
 		artCtl:  artifact.Ctl,
 		proMgr:  project.Mgr,
-		repoMgr: repository.Mgr,
+		repoCtl: repository.Ctl,
 	}
 }
 
 type artifactAPI struct {
 	BaseAPI
 	artCtl  artifact.Controller
 	proMgr  project.Manager
-	repoMgr repository.Manager
+	repoCtl repository.Controller
+}
+
+// TODO immutable, quota, readonly middlewares should cover this API
+func (a *artifactAPI) CopyArtifact(ctx context.Context, params operation.CopyArtifactParams) middleware.Responder {
+	if err := a.RequireProjectAccess(ctx, params.ProjectName, rbac.ActionCreate, rbac.ResourceArtifact); err != nil {
+		return a.SendError(ctx, err)
+	}
+	srcRepo, srcRef, err := parse(params.From)
```

Comment:
The API allows copy from artifact in other project, is there any security check?